src/MongoDB.Driver/FindOptions.cs

```diff
@@ -135,6 +179,33 @@ public BsonDocument Modifiers
             get { return _oplogReplay; }
             set { _oplogReplay = value; }
         }
+
+        /// <summary>
+        /// Gets or sets returnKey. If true, returns only the index keys in the resulting documents.
+        /// </summary>
+        public bool? ReturnKey
+        {
+            get { return _returnKey; }
+            set { _returnKey = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets whether the record Id should be added to the result document.
+        /// </summary>
+        public bool? ShowRecordId
+        {
+            get { return _showRecordId; }
+            set { _showRecordId = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets whether to use snapshot behavior.
+        /// </summary>
+        public bool? Snapshot
```

Comment:
`Snapshot` is `Obsolete` in `FindOperation`. Not sure if I should deprecate it here as well

Reply:
Maybe find out why Snapshot is obsolete?

It would be unusual for a new property to be obsolete from the moment it was added...

Reply:
It was deprecated in server version 3.6.1 and removed in 3.7.4
Here is the ticket: https://jira.mongodb.org/browse/CSHARP-2214, respective drivers (https://jira.mongodb.org/browse/DRIVERS-454) and spec (https://jira.mongodb.org/browse/SPEC-1033) tickets.
The reason for my confusion is same as for `MaxScan`. I will update the code shortly.